using(...)
{
    try
    {

    }
    catch(exception e)
    {


    }
}

跟 
try
{
    using(...)
    {
        ...
    }
}
catch(exception e)
{
    ...
}

首先要知道using 可以幫助直接執行Dispose的動作
第一種方法，捕捉到的例外，會是執行時後的情況
第二種方法，捕捉到Dispose的例外，如果Dispose有例外情況的話，如果Dispose地方沒有例外，也是會捕捉到執行例外


底層不處理錯誤，往前端傳遞，但訊息要經過包裝。